megamek/src/megamek/client/ui/swing/util/UIUtil.java

```diff
@@ -0,0 +1,735 @@
+     */
+    public static Color uiC3Color() {
+        return uiLightViolet();
+    }
+    
+    /** 
+     * Returns a color for the UI display of C3 Info. Different 
+     * colors will be supplied for a dark and for a light UI look-and-feel.
+     */
+    public static Color uiNickColor() {
+        return uiLightGreen();
+    }
+    
+    /** 
+     * Returns a color for the UI display of C3 Info. Different 
+     * colors will be supplied for a dark and for a light UI look-and-feel.
+     */
+    public static Color uiTTWeaponColor() {
+        return uiLightBlue();
+    }
+    
+    public static int scaleForGUI(int value) {
+        return Math.round(scaleForGUI((float)value));
+    }
+    
+    public static float scaleForGUI(float value) {
+        return GUIPreferences.getInstance().getGUIScale() * value;
+    }
+    
+    public static Dimension scaleForGUI(Dimension dim) {
+        float scale = GUIPreferences.getInstance().getGUIScale();
+        return new Dimension((int)(scale * dim.width), (int)(scale * dim.height));
+    }
+    
+    /** 
+     * Returns the provided color with its alpha value set to the provided alpha.
+     * alpha should be from 0 to 255 with 0 meaning transparent. 
+     */
+    public static Color addAlpha(Color color, int alpha) {
+        Objects.requireNonNull(color);
+        if (alpha < 0 || alpha > 255) {
+            throw new IllegalArgumentException("Alpha value out of range: " + alpha);
+        }
+        return new Color(color.getRed(), color.getGreen(), color.getBlue(), alpha);
+    }
+    
+    /** 
+     * Returns a grayed-out version of the given color. gray should be from 0 to 255
+     * with 255 meaning completely gray. Does not change the brightness, nor alpha. 
+     */
+    public static Color addGray(Color color, int gray) {
+        Objects.requireNonNull(color);
+        if (gray < 0 || gray > 255) {
+            throw new IllegalArgumentException("Gray value out of range: " + gray);
+        }
+        int mid = (color.getRed() + color.getGreen() + color.getBlue()) * gray / 3;
+        int red = (color.getRed() * (255 - gray) + mid) / 255;
+        int green = (color.getGreen() * (255 - gray) + mid) / 255;
+        int blue = (color.getBlue() * (255 - gray) + mid) / 255;
+        return new Color(red, green, blue, color.getAlpha());
+    }
+    
+    /** Returns the given String str enclosed in HTML tags and with a font tag according to the guiScale. */ 
+    public static String scaleStringForGUI(String str) {
+        return "<HTML>" + UIUtil.guiScaledFontHTML() + str + "</FONT></HTML>";
+    }
+    
+    /** Returns the given String str enclosed in HTML tags and with a font tag according to the guiScale. */ 
+    public static String scaleMessageForGUI(String str) {
+        return "<HTML>" + UIUtil.guiScaledFontHTML() + Messages.getString(str) + "</FONT></HTML>";
+    }
+    
+    public static void adjustDialog(Container contentPane) {

```

Comment:
What's this do?

Reply:
This applies the current gui scale to a dialog by setting fonts and adjusting EmptyBorders. This works kinda well with simple dialogs (the host game dialog, for example).